### PR TITLE
Add automatic verification tests

### PR DESCRIPTION
**Pull request by mikecirioli**
WIP

To test locally, set `GCP_PROJECT` to something suitable, then :

`make core-oc; make core-mm; make nginx; make deployer; make ubbagent; make install-app-crd`

which should ensure that all the needed images are available in your registry.  Then you can run the test like this:

```
mpdev verify --deployer=gcr.io/cirioli-marketplace-dev/cloudbees-core-billable/deployer@sha256:191101815db7419e876efc85f4ac55d4956eb081a67a72e44798aacbae0b66c4`
```

You will want to replace the deployer image with the location and sha256 hash of the one you just pushed in the first step.  If you make any changes to the deployer bits, you will need to re-run `make deployer` and update the sha256 hash of the image used by `mpdev`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#GCP_PROJECT=cloud-bill-dev
+GCP_PROJECT=cjp-marketplace-dev
+
```
Comment (schottsfired) [inline Makefile]: These lines shouldn't be necessary, since `GCP_PROJECT` is passed in as an env var.

File: Makefile [hunk 2]
```diff
@@ -120,7 +123,7 @@ install: install-app-crd deployer check-license-params
 	--parameters='{"name": "$(NAME)", "namespace": "$(NAMESPACE)", "numberOfUsers": "$(NUMBER_OF_USERS)", \
 	"customerFirstName": "$(CUSTOMER_FIRST_NAME)", "customerLastName": "$(CUSTOMER_LAST_NAME)", \
 	"customerEmail": "$(CUSTOMER_EMAIL)", "customerCompany": "$(CUSTOMER_COMPANY)", \
-	"reportingSecret": "gs://cloud-marketplace-tools/reporting_secrets/fake_reporting_secret.yaml"}'
+	"reportingSecret": ""}'
```
Comment (schottsfired) [inline Makefile]: What's wrong with using the fake secret?

File: deployer/deploy_with_tests.sh [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+if grep -q '$loadBalancerIp' "/data/nginx.yaml" ; then
+  deploy_gke_cloud
+else
+  deploy_gke_op
+fi
+
+sleep 180
+
+# add some actual tests here
```
Comment (schottsfired) [inline deployer/deploy_with_tests.sh]: Maybe. It'll work, but the recommendation is to add a "tester" image to `schema.yaml`.

File: deployer/deploy_with_tests.sh [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  --status "Pending"
+
+
+if grep -q '$loadBalancerIp' "/data/nginx.yaml" ; then
+  deploy_gke_cloud
+else
+  deploy_gke_op
+fi
+
+sleep 180
```
Comment (schottsfired) [inline deployer/deploy_with_tests.sh]: Better to check the health of the system than wait and assume everything is fine.

**Comment by schottsfired** [comment]
> ensure that all the needed images are available in your registry.

This isn't strictly necessary since `schema.yaml` is referencing other images: https://github.com/cloudbees/core-google-launcher/pull/18/files#diff-c45d6f437c9e5e8e700d637d8aa1b8a1R24

File: Makefile (hunk 1 above)
Comment (mikecirioli) [reply Makefile]: good catch

File: Makefile (hunk 2 above)
Comment (mikecirioli) [reply Makefile]: hmm, i thought i had left that in...

File: deployer/deploy_with_tests.sh (hunk 3 above)
Comment (mikecirioli) [reply deployer/deploy_with_tests.sh]: is that where the actual test harness is expected to run from?  figuring that part out was next on my list

File: deployer/deploy_with_tests.sh (hunk 4 above)
Comment (mikecirioli) [reply deployer/deploy_with_tests.sh]: 👍

**Pull request merged**